app/jsx/pages/BrowsePage.jsx

```diff
@@ -60,6 +64,90 @@ felis tellus mollis orci, sed start a journal nunc eget odio.
       <br/><br/>
     </div>
   )}
+
+}
+
+// Tree traversal help via http://codepen.io/anon/pen/Ftkln
+// 'depts' property (JSON)
+// [ ...,
+//  {"id"=>"international",
+//   "name"=>"UCLA International Institute",
+//   "children"=>
+//    [{"id"=>"asia",
+//      "name"=>"UCLA Asia Institute",
+//      "children"=>[{"id"=>"ccs", "name"=>"UCLA Center for Chinese Studies"}]}, .. ], } .. ]
+class DeptTree extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { depts: this.props.depts }
+    // this.onSelect = this.onSelect.bind(this)
+  }
+  onSelect(node) {
```

Comment:
Martin, I did not think through the onSelect at all. We can remove it for now if you like.